java/semantickernel-api/src/main/java/com/microsoft/semantickernel/plugin/KernelPluginCollection.java

```diff
@@ -1,21 +1,48 @@
 package com.microsoft.semantickernel.plugin;
 
-import java.util.Iterator;
-
 import com.microsoft.semantickernel.orchestration.KernelFunction;
-import com.microsoft.semantickernel.plugin.KernelPlugin;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
 
 public class KernelPluginCollection implements Iterable<KernelPlugin> {
 
+    private final Map<String, KernelPlugin> plugins = new HashMap<>();
```

Comment:
Somewhere else we have a collection that implements Map. Why should this be different? Any collection class we have should have a common interface.